pkg/cmd/membership/membership.go

```diff
@@ -0,0 +1,168 @@
+package membership
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/coreos/etcd/clientv3"
+	"github.com/coreos/etcd/pkg/transport"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog"
+
+	"github.com/openshift/cluster-etcd-operator/pkg/cmd/waitforkube"
+)
+
+const (
+	retryDuration     = 2 * time.Second
+	dialTimeout       = 5 * time.Second
+	etcdCertFile      = "/var/run/secrets/etcd-client/tls.crt"
+	etcdKeyFile       = "/var/run/secrets/etcd-client/tls.key"
+	etcdTrustedCAFile = "/var/run/configmaps/etcd-ca/ca-bundle.crt"
+	dbPath            = "member/snap/db"
+)
+
+type membership struct {
+	errOut         io.Writer
+	EtcdDnsName    string
+	EtcdEndpoints  string
+	etcdDataDir    string
+	EtcdDataFile   string
```

Comment:
does this have to be exported variable?

Reply:
can we have a function instead that calculates data file from data directory and dbfile constant? I think that we make unit testing this easier

Reply:
> does this have to be exported variable?

To keep this simple it needs to be exported because the value is populated in Complete() after struct is initialized. We could create another struct and initialize it but it seemed a bit much for this little script so I just used membership.

>can we have a function instead that calculates data file from data directory and dbfile constant? I think that will make unit testing this easier

OK we can do that but the field is exported so that it could be injected directly as well right?